tldw_Server_API/app/core/Chat/chat_service.py

```
@@ -285,28 +285,136 @@ def _discover_openrouter_models_for_chat(*, force_refresh: bool = False) -> tupl
+    requested_character_id = getattr(request_data, "character_id", None) if request_data is not None else None
+    if requested_character_id:
+        return "regular" if _is_default_character_reference(
+            requested_character_id,
+            default_character_id=default_character_id,
+        ) else "character"
+
+    return "regular"
+
+
+async def _resolve_default_character_id(
+    chat_db: Any,
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `chat_db`

(ANN401)